src/preferences/colorpaletteeditormodel.cpp

```diff
@@ -1,15 +1,112 @@
 #include "preferences/colorpaletteeditormodel.h"
 
+#include <util/assert.h>
+
+#include <QList>
+#include <QMap>
+#include <QMultiMap>
+#include <QRegularExpression>
+#include <algorithm>
+
+#include "engine/controls/cuecontrol.h"
```

Comment:
including the entire large header file (whose contents are largely unrelated) just for `NUM_HOT_CUES` is probably overkill. `NUM_HOT_CUES` should probably be factored out into a smaller header file and renamed `static constexpr int mixxx::kNumHotCues`, wdyt?

Reply:
static in this context is nothing else than a anonymous namespace. There where efforts to deprecate global statics in favor of this but that has not made it's way into the C++ standard. In Mixxx we prefer the anonymous namespace.

Reply:
> In Mixxx we prefer the anonymous namespace.

True, but anonymous namespaces are only used in `*.cpp` files, not header files.

Reply:
A header file is nothing special in cpp. It is just copied into the cpp file in the preprocessor stage.
In case the global variable is constexpr it is part of the text segment. This is also the case for static const or const in an anonymous namespace, where the compiler does not need to expose the address to the linker. 

If you have an anonymous namespace in a header and you include the header in two different cpp files, you have two separate variables. The compiler is free to duplicate the value even more for performance reasons.

If the variable is not a constexpr or plane old data type, the variable is initialized when accessed the first time. 
This happens for all our QStrings we have there.

Reply:
I know, but he point is that an anonymous namespace tells the compiler that everything inside it is local to the compilation unit it is found in. 

If you put an anonymous namespace into a header file, you need to use every single function or variable inside that namespace in each cpp file, otherwise you'll get a bunch of "unused variable" warnings. And for functions this will make the executable larger due to redundant code (unless the compiler deduplicates the redundant function at link time).

Therefore, anonymous namespace don't make sense in header files.

Reply:
Yes, however this applied to static const and to namespaces in the same way. The workaround is to use macros.